File: pkg/codegen/testing/test/testdata/third-party-package-pp/dotnet/third-party-package.cs
using System.Collections.Generic;
using Pulumi;
using Other = ThirdParty.Other;

return await Deployment.RunAsync(() => 
{
    var Other = new Other.Thing("Other", new()
    {
        Idea = "Support Third Party",
    });

    var Question = new Other.Module.Object("Question", new()
    {
        Answer = 42,
    });

    var Provider = new Other.Provider("Provider", new()
    {
        ObjectProp = 
        {
            { prop1 = "foo" },
            { prop2 = "bar" },
            { prop3 = "fizz" },
        },
    });

});

File: pkg/codegen/testing/test/testdata/third-party-package-pp/go/third-party-package.go
package main

import (
	"git.example.org/thirdparty/sdk/go/pkg"
	"git.example.org/thirdparty/sdk/go/pkg/module"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := pkg.NewThing(ctx, "Other", &pkg.ThingArgs{
			Idea: "Support Third Party",
		})
		if err != nil {
			return err
		}
		_, err = module.NewObject(ctx, "Question", &module.ObjectArgs{
			Answer: 42,
		})
		if err != nil {
			return err
		}
		_, err = module.NewObject(ctx, "Question2", &module.ObjectArgs{
			Answer: 24,
		})
		if err != nil {
			return err
		}
		_, err = pkg.NewProvider(ctx, "Provider", &pkg.ProviderArgs{
			ObjectProp: map[string]pulumi.String{
				"prop1": pulumi.String("foo"),
				"prop2": pulumi.String("bar"),
				"prop3": pulumi.String("fizz"),
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}
